src/pfe/file-watcher/server/src/utils/kubeutil.ts

```diff
@@ -333,6 +333,98 @@ export async function installChart(projectID: string, deploymentName: string, ch
     return response;
 }
 
+export async function exposeOverIngress(projectID: string, projectName: string, ingressDomain: string, isHTTPS: boolean): Promise<string> {
+    let ownerReferenceName: string;
+    let ownerReferenceUID: string;
+    let serviceName: string;
+    let servicePort: number;
+
+    try {
+        let resp: any = undefined;
```

Comment:
D'oh, good catch. I'll fix that

Reply:
Actually, since we'll be refreshing the ingress each time, this shouldn't be needed.

I did change the code that retrieves the service and deployment so that we would error out if it couldn't be found, as we'd run into issues trying to create an ingress pointing to a blank service.

Reply:
Ok, makes sense. Can we remove the `deployments.get` call though? It looks like we're only using it to get the ownerReferenceName and ownerReferenceUID for which both should be available from the service which is the next call.